xclim/sdba/detrending.py

```diff
@@ -90,47 +126,33 @@ class PolyDetrend(BaseDetrend):
 
     Parameters
     ----------
+    group : Union[str, Grouper]
+        The grouping information. See :py:class:`xclim.sdba.base.Grouper` for details.
+        The fit is performed along the group's main dim.
+    kind : {'*', '+'}
+        The way the trend is removed or added, either additive or multiplicative.
     degree : int
-      The order of the polynomial to fit.
-    freq : Optional[str]
-      If given, resamples the data to this frequency before computing the trend.
-    kind : {'+', '*'}
-      The way the trend is removed and put back, either additively or multiplicatively.
-
-    Notes
-    -----
-    If freq is used to resample at a lower frequency, make sure the series includes full periods.
+        The order of the polynomial to fit.
+    preserve_mean : bool
+        Whether to preserve the mean when de/re-trending. If True, the trend has its mean
+        removed before it is used.
     """
 
-    def __init__(self, degree=4, freq=None, kind=ADDITIVE, preserve_mean=False):
+    def __init__(self, group="time", kind=ADDITIVE, degree=4, preserve_mean=False):
         super().__init__(
-            degree=degree, freq=freq, kind=kind, preserve_mean=preserve_mean
+            group=group, kind=kind, degree=degree, preserve_mean=preserve_mean
         )
 
     def _fit(self, da, dim="time"):
-        if self.freq is not None:
-            da = da.resample(
-                time=self.freq, label="left", loffset=loffsets[self.freq]
-            ).mean()
-        self._fitds = da.polyfit(dim=dim, deg=self.degree, full=True)
+        return da.polyfit(dim=dim, deg=self.degree)
 
-    def _get_trend(self, da):
+    def _get_trend(self, grpd, dim="time"):
```

Comment:
I remember hard-coding `time` in a couple of places in the code. Probably worth writing a test with a `mytime` dimension to check everything work as expected.

Reply:
Good idée